pkg/apis/autoscaling/v2beta1/conversion.go

```diff
@@ -247,8 +247,10 @@ func Convert_autoscaling_ObjectMetricStatus_To_v2beta1_ObjectMetricStatus(in *au
 	}
 	out.MetricName = in.Metric.Name
 	out.Selector = in.Metric.Selector
-	currentAverageValue := *in.Current.AverageValue
-	out.AverageValue = &currentAverageValue
+	if in.Current.AverageValue != nil {
```

Comment:
needs a test demonstrating the bug and that this fixes the issue

Reply:
does this have the same issue:

https://github.com/kubernetes/kubernetes/blob/a0c2788249ae1582d10089e7a34bb54fc6b3879d/pkg/apis/autoscaling/v2beta1/conversion.go#L188-L189